File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/VariantBridge.kt
import com.superwall.sdk.models.triggers.Experiment
import com.superwall.sdk.models.triggers.Experiment.Variant

fun Experiment.Variant.toJson(): Map<String, Any?> {
    return mapOf(
        "id" to id,
        "type" to type.toJson(),
        "paywallId" to paywallId
    )
}

fun Experiment.Variant.VariantType.toJson(): String {
    return when (this) {
        Variant.VariantType.TREATMENT -> "treatment"
        Variant.VariantType.HOLDOUT -> "holdout"
    }
}

File: ios/Classes/Bridges/VariantBridge.swift
import Flutter
import SuperwallKit

extension Experiment.Variant {
  func toJson() -> [String: Any] {
    var dict = [String: Any]()
    dict["id"] = id
    dict["type"] = type.toJson()
    dict["paywallId"] = paywallId
    return dict
  }
}

extension Experiment.Variant.VariantType {
  func toJson() -> String {
    switch self {
      case .treatment:
        return "treatment"
      case .holdout:
        return "holdout"
    }
  }
}

